### PR TITLE
cleanup(pkg): switch to use master go-github to get rid of internal variables API client

### PR DESCRIPTION
**Pull request by FedeDP**
**What type of PR is this?**

/kind cleanup

**Any specific area of the project related to this PR?**

/area pkg
/area tests

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: main_test.go [hunk 1]
```diff
@@ -25,125 +23,31 @@ orgs:
             - secret2
 `
 
-type MockVariableService struct {
```
Comment (FedeDP) [inline main_test.go:28]: Moved to `poiana.variables_mock`

File: main_test.go [hunk 2]
```diff
@@ -25,125 +23,31 @@ orgs:
...
-	_ = mServ.CreateOrUpdateRepoVariable(context.Background(), "", "", &poiana.Variable{
-		Name:      "var0",
-		Value:     "value0",
-		CreatedAt: github.Timestamp{},
-		UpdatedAt: github.Timestamp{},
-	})
-	return mServ
-}
-
-type MockSecretsService struct {
```
Comment (FedeDP) [inline main_test.go:69]: Moved to `poiana.secrets_mock`

File: main_test.go [hunk 3]
```diff
@@ -25,125 +23,31 @@ orgs:
...
-func newMockSecretsService() poiana.ActionsSecretsService {
-	mServ := &MockSecretsService{secrets: make(map[string]*github.EncryptedSecret, 0)}
-	_ = mServ.CreateOrUpdateRepoSecret(context.Background(), "", "", &github.EncryptedSecret{
-		Name:  "secret0",
-		KeyID: "testing",
-	})
-	return mServ
-}
-
-type MockPublicKeyProvider struct{}
```
Comment (FedeDP) [inline main_test.go:106]: Moved to `poiana.secrets_mock`

File: pkg/poiana/client.go [hunk 4]
```diff
@@ -50,12 +22,5 @@ func NewClient(ctx context.Context, tokenFile string) (*Client, error) {
...
 	}
-	ghCl := github.NewClient(tc)
-	return &Client{
-		Client: ghCl,
-		Actions: &actionsService{
-			ActionsService: ghCl.Actions,
-			client:         ghCl,
-		},
-	}, nil
+	return github.NewClient(tc), nil
```
Comment (FedeDP) [inline pkg/poiana/client.go:25]: We could possibly move this function to main.go and delete this whole file

File: pkg/poiana/secrets.go [hunk 5]
```diff
@@ -14,26 +14,11 @@ type SecretsProvider interface {
 }
 
 type PublicKeyProvider interface {
-	GetPublicKey(context.Context, string, string) (*github.PublicKey, error)
+	GetRepoPublicKey(ctx context.Context, orgName string, repoName string) (*github.PublicKey, *github.Response, error)
```
Comment (FedeDP) [inline pkg/poiana/secrets.go:17]: Now internal interfaces exactly match github package methods.

File: pkg/poiana/client.go (hunk 4 above)
Comment (FedeDP) [reply pkg/poiana/client.go:25]: For now, it is just an helper in case we want to add new functionalities.

**Review by jasondellaluce — APPROVED** [review]

**Pull request merged**